internal/pkg/aws/ec2/ec2.go

```diff
@@ -52,6 +59,26 @@ func New(s *session.Session) *EC2 {
 	}
 }
 
+// ListVpcSubnets lists all subnets given a VPC ID.
+func (c *EC2) ListVpcSubnets(vpcID string) (*Subnets, error) {
+	respSubnets, err := c.subnets(Filter{
+		Name:   "vpc-id",
+		Values: []string{vpcID},
+	})
+	if err != nil {
+		return nil, err
+	}
+	var subnets Subnets
+	for _, subnet := range respSubnets {
+		if aws.BoolValue(subnet.MapPublicIpOnLaunch) {
+			subnets.PublicSubnets = append(subnets.PublicSubnets, aws.StringValue(subnet.SubnetId))
```

Comment:
_technically_ these subnets need an IG to be considered public, but I think publicIP on launch is a good approximation.

Reply:
yeah good to know!